bindings/netstandard/ElectionGuard/ElectionGuard.Decryption/Verify/VerifyElection.cs

```diff
@@ -0,0 +1,1087 @@
+    /// (4.5) c = H(04,Q;K,α,β,a0,b0,a1,b1).
+    /// An election verifier must then confirm the following:
+    /// (4.A) The given values α and β are in the set Zrp. 
+    ///       (A value x is in Zrp if and only if x is an integer such that 0 ≤ x < p and xq mod p = 1 is satisfied.)
+    /// (4.B) The given values c0, c1, v0, and v1 are each in the set Zq. 
+    ///       (A value x is in Zq if and only if x is an integer such that 0 ≤ x < q.)
+    /// (4.C) The equation c = (c0 + c1) mod q is satisfied.
+    /// </summary>
+    private static Task<VerificationResult> VerifyDisjunctiveProof(
+        ElectionConstants constants,
+        CiphertextElectionContext context,
+        CiphertextBallotSelection selection
+    )
+    {
+        var results = new List<VerificationResult>();
+
+        // Verification 4.1 - 𝑎0 = 𝑔^𝑣0 mod 𝑝 ⋅ 𝛼^𝑐0 mod 𝑝
+        using var gv0 = BigMath.PowModP(constants.G, selection.Proof.ZeroResponse);
+        using var ac0 = BigMath.PowModP(selection.Ciphertext.Pad, selection.Proof.ZeroChallenge);
+        using var gv0ac0 = BigMath.MultModP(gv0, ac0);
+        var consistent_gv0 = selection.Proof.ZeroPad.Equals(gv0ac0);
+        results.Add(new VerificationResult(consistent_gv0, $"      - Verification 4.1: 𝑎0 = 𝑔^𝑣0 mod 𝑝 ⋅ 𝛼^𝑐0 mod 𝑝"));
+
+        // Verification 4.2 - 𝑎1 = 𝑔^𝑣1 mod 𝑝 ⋅ 𝛼^𝑐1 mod 𝑝
+        using var gv1 = BigMath.PowModP(constants.G, selection.Proof.OneResponse);
+        using var ac1 = BigMath.PowModP(selection.Ciphertext.Pad, selection.Proof.OneChallenge);
+        using var gv1ac1 = BigMath.MultModP(gv1, ac1);
+        var consistent_gv1 = selection.Proof.OnePad.Equals(gv1ac1);
+        results.Add(new VerificationResult(consistent_gv1, $"      - Verification 4.2: 𝑎1 = 𝑔^𝑣1 mod 𝑝 ⋅ 𝛼^𝑐1 mod 𝑝"));
+
+        // Verification 4.3 - 𝑏0 = 𝐾^𝑣0 mod 𝑝 ⋅ 𝛽^𝑐0 mod 𝑝
+        using var kv0 = BigMath.PowModP(context.ElGamalPublicKey, selection.Proof.ZeroResponse);
+        using var bc0 = BigMath.PowModP(selection.Ciphertext.Data, selection.Proof.ZeroChallenge);
+        using var kv0bc0 = BigMath.MultModP(kv0, bc0);
+        var consistent_kv0 = selection.Proof.ZeroData.Equals(kv0bc0);
+        results.Add(new VerificationResult(consistent_kv0, $"      - Verification 4.3: 𝑏0 = 𝐾^𝑣0 mod 𝑝 ⋅ 𝛽^𝑐0 mod 𝑝"));
+
+        // Verification 4.4 - 𝑏1 = 𝐾^w1 mod 𝑝 ⋅ 𝛽^𝑐1 mod 𝑝
+        using var w1 = BigMath.SubModQ(selection.Proof.OneResponse, selection.Proof.OneChallenge);
+        using var kw1 = BigMath.PowModP(context.ElGamalPublicKey, w1);
+        using var bc1 = BigMath.PowModP(selection.Ciphertext.Data, selection.Proof.OneChallenge);
+        using var kw1bc1 = BigMath.MultModP(kw1, bc1);
+        var consistent_kw1 = selection.Proof.OneData.Equals(kw1bc1);
+        results.Add(new VerificationResult(consistent_kw1, $"      - Verification 4.4: 𝑏1 = 𝐾^w1 mod 𝑝 ⋅ 𝛽^𝑐1 mod 𝑝"));
+
+        // Verification 4.5 - c = H(04,Q,K,α,β,a0,b0,a1,b1)
+        // using var recomputedHash = Hash.HashElems(
```

Comment:
theoretically this hash could validate so i left the code commented out here but i was running into issues with it. it's either going to be fixed as part of review or it will be broken out into a new issue